File: src/main/java/net/guizhanss/ultimategenerators2/core/services/ConfigurationService.java
package net.guizhanss.ultimategenerators2.core.services;

import net.guizhanss.guizhanlib.slimefun.addon.AddonConfig;
import net.guizhanss.ultimategenerators2.UltimateGenerators2;

import lombok.AccessLevel;
import lombok.Getter;

@Getter
public final class ConfigurationService {
    @Getter(AccessLevel.NONE)
    private final AddonConfig config;

    private boolean autoUpdate;
    private boolean debug;
    private String lang;
    private int glassElectricityTransmitterCraftAmount;

    public ConfigurationService(UltimateGenerators2 plugin) {
        config = new AddonConfig(plugin, "config.yml");
        reload();
    }

    public void reload() {
        config.reload();
        config.addMissingKeys();

        autoUpdate = config.getBoolean("auto-update", true);
        debug = config.getBoolean("debug", false);
        lang = config.getString("lang", "en");
        glassElectricityTransmitterCraftAmount = config.getInt("glass-electricity-transmitter.craft-amount", 12);

        config.save();
    }
}

File: src/main/kotlin/net/guizhanss/ultimategenerators2/core/services/ConfigurationService.kt
package net.guizhanss.ultimategenerators2.core.services

import net.guizhanss.guizhanlib.slimefun.addon.AddonConfig
import net.guizhanss.ultimategenerators2.UltimateGenerators2

class ConfigurationService(plugin: UltimateGenerators2) {
    private val config = AddonConfig(plugin, "config.yml")

    var autoUpdate = true
        private set
    var debug = false
        private set
    var lang = "en"
        private set
    var glassElectricityTransmitterCraftAmount = 12
        private set
    var enderCrystalGeneratorEnergyPerCrystal = 128
        private set

    init {
        reload()
    }

    fun reload() {
        config.reload()
        config.addMissingKeys()

        autoUpdate = config.getBoolean("auto-update", true)
        debug = config.getBoolean("debug", false)
        lang = config.getString("lang", "en")!!
        glassElectricityTransmitterCraftAmount = config.getInt("glass-electricity-transmitter.craft-amount", 12)
        enderCrystalGeneratorEnergyPerCrystal = config.getInt("ender-crystal-generator.energy-per-crystal", 128)

        config.save()
    }
}
